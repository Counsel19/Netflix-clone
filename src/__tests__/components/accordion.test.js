/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/prefer-presence-queries */
/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import faqsData from "../../fixtures/faqs.json";
import { Accordion } from "../../components";

describe("<Accordion />", () => {
  it("renders the accordion with populated data", () => {
    const { container, getByText } = render(
      <Accordion>
        <Accordion.Title>Frequestly Asked Questions</Accordion.Title>

        {faqsData.map((item) => {
          return (
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          );
        })}
      </Accordion>
    );

    expect(getByText("Frequestly Asked Questions")).toBeTruthy();
    expect(getByText("What is Netflix?")).toBeTruthy();
    expect(getByText("How much does Netflix cost?")).toBeTruthy();
    expect(getByText("Where can I watch?")).toBeTruthy();
    expect(getByText("How do I cancel?")).toBeTruthy();
    expect(getByText("What can I watch on Netflix?")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Opens and closes the <Accordion /> component", async () => {
    const { container, queryByText} = render(
      <Accordion>
        <Accordion.Title>Frequestly Asked Questions</Accordion.Title>

        {faqsData.map((item) => {
          return (
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          );
        })}
      </Accordion>
    );

    const whatIsNetflixBodyText = "Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices. You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!";

      expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();
      fireEvent.click(queryByText('What is Netflix?'));
      expect(queryByText(whatIsNetflixBodyText)).toBeTruthy();

      fireEvent.click(queryByText('What is Netflix?'));
      expect(container.firstChild).toMatchSnapshot()

  });
});
