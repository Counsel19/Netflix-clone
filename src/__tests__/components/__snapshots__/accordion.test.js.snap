// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> Opens and closes the <Accordion /> component 1`] = `
<div
  class="accordion__Container-sc-fhykpk-0 hNHqeY"
>
  <div
    class="accordion__Inner-sc-fhykpk-1 iLsKYO"
  >
    <h1
      class="accordion__Title-sc-fhykpk-2 hyWijK"
    >
      Frequestly Asked Questions
    </h1>
    <div
      class="accordion__Item-sc-fhykpk-3 gMjtLQ"
    >
      <div
        class="accordion__Header-sc-fhykpk-4 hXRjwk"
      >
        What is Netflix?
        <img
          alt="Add"
          src="./images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-fhykpk-3 gMjtLQ"
    >
      <div
        class="accordion__Header-sc-fhykpk-4 hXRjwk"
      >
        How much does Netflix cost?
        <img
          alt="Add"
          src="./images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-fhykpk-3 gMjtLQ"
    >
      <div
        class="accordion__Header-sc-fhykpk-4 hXRjwk"
      >
        Where can I watch?
        <img
          alt="Add"
          src="./images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-fhykpk-3 gMjtLQ"
    >
      <div
        class="accordion__Header-sc-fhykpk-4 hXRjwk"
      >
        How do I cancel?
        <img
          alt="Add"
          src="./images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-fhykpk-3 gMjtLQ"
    >
      <div
        class="accordion__Header-sc-fhykpk-4 hXRjwk"
      >
        What can I watch on Netflix?
        <img
          alt="Add"
          src="./images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders the accordion with populated data 1`] = `
<div
  class="accordion__Container-sc-fhykpk-0 hNHqeY"
>
  <div
    class="accordion__Inner-sc-fhykpk-1 iLsKYO"
  >
    <h1
      class="accordion__Title-sc-fhykpk-2 hyWijK"
    >
      Frequestly Asked Questions
    </h1>
    <div
      class="accordion__Item-sc-fhykpk-3 gMjtLQ"
    >
      <div
        class="accordion__Header-sc-fhykpk-4 hXRjwk"
      >
        What is Netflix?
        <img
          alt="Add"
          src="./images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-fhykpk-3 gMjtLQ"
    >
      <div
        class="accordion__Header-sc-fhykpk-4 hXRjwk"
      >
        How much does Netflix cost?
        <img
          alt="Add"
          src="./images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-fhykpk-3 gMjtLQ"
    >
      <div
        class="accordion__Header-sc-fhykpk-4 hXRjwk"
      >
        Where can I watch?
        <img
          alt="Add"
          src="./images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-fhykpk-3 gMjtLQ"
    >
      <div
        class="accordion__Header-sc-fhykpk-4 hXRjwk"
      >
        How do I cancel?
        <img
          alt="Add"
          src="./images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-fhykpk-3 gMjtLQ"
    >
      <div
        class="accordion__Header-sc-fhykpk-4 hXRjwk"
      >
        What can I watch on Netflix?
        <img
          alt="Add"
          src="./images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;
